ocs_ci/ocs/longevity.py

```diff
@@ -555,6 +558,255 @@ def stage_0(
+                                            list.
+        teardown_factory : Fixture to tear down a resource that was created during the test.
+        num_of_pvcs (int) : Total Number of PVCs we want to create.
+        pvc_size (int) : Size of each PVC in GB.
+        run_time (int) : Total Run Time in minutes.
+        pvc_size_new (int) : Size of the expanded PVC in GB.
+
+    """
+    end_time = datetime.now() + timedelta(minutes=run_time)
+    cycle_no = 0
+
+    while datetime.now() < end_time:
+        cycle_no += 1
+        log.info(f"------------STARTING CYCLE:{cycle_no}------------")
+
+        namespace = "stage-4-cycle-" + str(cycle_no)
+        project = project_factory(project_name=namespace)
+        executor = ThreadPoolExecutor(max_workers=1)
+        pvc_objs = list()
+
+        for interface in (constants.CEPHFILESYSTEM, constants.CEPHBLOCKPOOL):
+            if interface == constants.CEPHFILESYSTEM:
+                access_modes = [constants.ACCESS_MODE_RWO, constants.ACCESS_MODE_RWX]
+                num_of_pvc = num_of_pvcs // 2
+            else:
+                access_modes = [
+                    constants.ACCESS_MODE_RWO,
+                    constants.ACCESS_MODE_RWO + "-" + constants.VOLUME_MODE_BLOCK,
+                    constants.ACCESS_MODE_RWX + "-" + constants.VOLUME_MODE_BLOCK,
+                ]
+                num_of_pvc = num_of_pvcs - num_of_pvcs // 2
+
+            # Create PVCs
+            if num_of_pvc > 0:
+                pvc_objs_tmp = multi_pvc_factory(
+                    interface=interface,
+                    size=pvc_size,
+                    project=project,
+                    access_modes=access_modes,
+                    status=constants.STATUS_BOUND,
+                    num_of_pvc=num_of_pvc,
+                    wait_each=True,
+                )
+                log.info("PVC creation was successful.")
+                pvc_objs.extend(pvc_objs_tmp)
+
+            else:
+                log.info(
+                    f"Num of PVCs of interface - {interface} = {num_of_pvc}. So no PVCs created."
+                )
+
+        # Create PODs
+        pod_objs = list()
+        for pvc_obj in pvc_objs:
+            if pvc_obj.get_pvc_vol_mode == constants.VOLUME_MODE_BLOCK:
+                pod_objs.append(
+                    pod_factory(
+                        pvc=pvc_obj,
+                        raw_block_pv=True,
+                        status=constants.STATUS_RUNNING,
+                        pod_dict_path=constants.PERF_BLOCK_POD_YAML,
+                    )
+                )
+            else:
+                pod_objs.append(
+                    pod_factory(
+                        pvc=pvc_obj,
+                        status=constants.STATUS_RUNNING,
+                        pod_dict_path=constants.PERF_POD_YAML,
+                    )
+                )
+
+        log.info("POD creation was successful.")
+
+        # Run IO to utilize 25% of volume
```

Comment:
The code from PVC creation till running the IO is handled in _multi_pvc_pod_lifecycle_factory in stage 2 PR . Lets use that function here once the stage2 PR is merged.

Reply:
Ack